### PR TITLE
All fields

### PR DESCRIPTION
**Pull request by itye-msft**
In this PR I have added a simple flag in protobuf template called `allFields` which allows to analyze all fields if it is set to True.

### CODE REVIEW

File: presidio-analyzer/analyzer/analyzer_engine.py [hunk 1]
```diff
@@ -117,6 +118,24 @@ def __convert_fields_to_entities(self, fields):
...
+        """
+        Returns a list of fields. If the "allFields" flag if True, all fields will be returned. Otherwise,
+        the template fields will be returned.
+
+        :param analyzeTemplate: The template to process.
+        :return: a list of fields
+        """
+        if analyzeTemplate.allFields:
+            fields = []
+            for item in common_pb2.FieldTypesEnum.keys():
```
Comment (omri374) [inline presidio-analyzer/analyzer/analyzer_engine.py]: This is not going to work for custom recognizers. the allFields param should skip the filtering of recognizers by field

File: presidio-analyzer/analyzer/analyzer_engine.py [hunk 2]
```diff
@@ -64,22 +65,25 @@ def Apply(self, request, context):
...
 
     def get_language_from_request(self, request):
-        language = request.analyzeTemplate.languageCode
+        language = request.analyzeTemplate.language
         if language is None or language == "":
             language = DEFAULT_LANGUAGE
         return language
 
-    def analyze(self, text, entities, language):
+    def analyze(self, text, entities, language, all_fields=False):
```
Comment (eladiw) [inline presidio-analyzer/analyzer/analyzer_engine.py]: default 'False' is not needed as the only place that calls this method is lie 57, which passes a value.
you probably need to update some tests after your remove the default value

File: presidio-analyzer/analyzer/recognizer_registry/recognizer_registry.py [hunk 3]
```diff
@@ -63,38 +63,40 @@ def remove_recognizer(self, name):
...
-
-            if len(subset) == 0:
-                logging.warning(
-                    "Entity " + entity +
-                    " doesn't have the corresponding recognizer in language :"
-                    + language)
-            else:
-                to_return.extend(subset)
+        if all_fields:
+            to_return = [rec for rec in self.recognizers if
```
Comment (eladiw) [inline presidio-analyzer/analyzer/recognizer_registry/recognizer_registry.py]: you can merge  this:

> [rec for rec in self.recognizers 
>  and language == rec.supported_language]

and then just filter for additional entities... (less duplicate code)

File: presidio-analyzer/analyzer/recognizer_registry/recognizer_registry.py [hunk 4]
```diff
@@ -63,38 +63,40 @@ def remove_recognizer(self, name):
...
+                         language == rec.supported_language]
+        else:
+            for entity in entities:
+                subset = [rec for rec in self.recognizers if
+                          entity in rec.supported_entities
+                          and language == rec.supported_language]
+
+                if len(subset) == 0:
+                    logging.warning(
```
Comment (eladiw) [inline presidio-analyzer/analyzer/recognizer_registry/recognizer_registry.py]: this line is going to fail linting once the other lint pr is merged .
go for:
"text text %s text", "text"
style

**Pull request merged**